src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1311,6 +1311,73 @@ void CodeGen::genAvxOrAvx2Intrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_AVX_SetAllVector256:
+        {
+            assert(op1 != nullptr);
+            assert(op2 == nullptr);
+            op1Reg = op1->gtRegNum;
+            if (varTypeIsIntegral(baseType))
+            {
+                // If the argument is a integer, it needs to be moved into a XMM register
+                regNumber tmpXMM = node->ExtractTempReg();
+                if (varTypeIsLong(baseType))
+                {
+                    emit->emitIns_R_R(INS_mov_i2xmm, emitTypeSize(baseType), tmpXMM, op1Reg);
+                }
+                else
+                {
+                    emit->emitIns_R_R(INS_mov_i2xmm, emitTypeSize(TYP_SIMD16), tmpXMM, op1Reg);
```

Comment:
Shouldn't this just always be `emitTypeSize(baseType)`?

Reply:
Unfortunately not... The current emitter for `movd`/`movq` is really messy, we need to refactor it later.

Reply:
Could you log a bug tracking this?

Reply:
Will do.

Reply:
Logged at https://github.com/dotnet/coreclr/issues/17051

Reply:
Here can be simplified, thanks for @mikedn 's help.